api/src/Serializer/Normalizer/RelatedCollectionLinkNormalizer.php

```diff
@@ -111,12 +114,17 @@ public function normalize($data, $format = null, array $context = []): null|arra
                 continue;
             }
 
-            try {
-                $normalized_data['_links'][$rel] = ['href' => $this->getRelatedCollectionHref($data, $rel, $context)];
-            } catch (UnsupportedRelationException $e) {
-                // The relation is not supported, or there is no matching filter defined on the related entity
+            // If relation is a public property, this property can be checked to be a non-null value
+            $values = get_object_vars($data);
+            if (array_key_exists($rel, $values) && null == $values[$rel]) {
+                // target-value is NULL
                 continue;
             }
+
+            list($ok, $href) = $this->getRelatedCollectionHref($data, $rel, $context);
```

Comment:
oh, we are programming go now

Reply:
would you prefer the following:
```php
$ok = $this->getRelatedCollectionHref($data, $rel, $context, out $href);
```

Reply:
oh no, am strictly against out parameters.
i was just surprised.
Exceptions ~= (on the same level) return codes > out (or worse: in/out) parameters.

In php its just not idiomatic... but the idiomatic exceptions are slow as shown here.

Reply:
it is ok, don't change it back after https://github.com/ecamp/ecamp3/pull/6536/commits/70a7155e779576c6723a15eb34ccff92bfafb192